nifi-commons/nifi-expression-language/src/main/java/org/apache/nifi/attribute/expression/language/evaluation/functions/GeohashStringEncodeEvaluator.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.attribute.expression.language.evaluation.functions;
+
+import org.apache.nifi.attribute.expression.language.EvaluationContext;
+import org.apache.nifi.attribute.expression.language.evaluation.Evaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.QueryResult;
+import org.apache.nifi.attribute.expression.language.evaluation.StringEvaluator;
+import org.apache.nifi.attribute.expression.language.evaluation.StringQueryResult;
+import org.apache.nifi.attribute.expression.language.exception.AttributeExpressionLanguageException;
+
+import ch.hsr.geohash.GeoHash;
+
+
+public class GeohashStringEncodeEvaluator extends StringEvaluator {
+
+    public enum GeohashStringFormat {
+        BASE32, BINARY
+    }
+
+    private final Evaluator<Number> latitude;
+    private final Evaluator<Number> longitude;
+    private final Evaluator<Long> level;
+    private final Evaluator<String> format;
+
+    public GeohashStringEncodeEvaluator(final Evaluator<Number> latitude, final Evaluator<Number> longitude, final Evaluator<Long> level, final Evaluator<String> format) {
+        this.latitude = latitude;
+        this.longitude = longitude;
+        this.level = level;
+        this.format = format;
+    }
+
+    @Override
+    public QueryResult<String> evaluate(final EvaluationContext evaluationContext) {
+        final Number latitudeValue = latitude.evaluate(evaluationContext).getValue();
+        if (latitudeValue == null) {
```

Comment:
Same as earlier comment. If we do not receive a valid value for latitude/longitude it seems as if no error/bulletin is appearing in NiFi. I think we should raise an exception, otherwise users would not be alerted in cases where data might be bad or corrupted.